src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1871,28 +1837,17 @@ internal static string GetRelativeDirectory(string path, string scriptRoot)
             return string.Empty;
         }
 
-        private static bool IsDisabled(JToken isDisabledValue, ScriptSettingsManager settingsManager)
+        public void ApplyJobHostMetadata()
```

Comment:
Just make this private and move the call from ScriptHostManager to this class OnHostInitialized as we discussed.